llvm/test/MC/Disassembler/AMDGPU/gfx11_dasm_vop1.txt

```diff
@@ -1,3644 +1,3646 @@
 
-# GFX11-FAKE16: v_ceil_f16_e32 v5, ttmp15               ; encoding: [0x7b,0xb8,0x0a,0x7e]
-# GFX11-REAL16: v_ceil_f16_e32 v5.l, ttmp15             ; encoding: [0x7b,0xb8,0x0a,0x7e]
 0x7b,0xb8,0x0a,0x7e
+# GFX11-REAL16: v_ceil_f16_e32 v5.l, ttmp15             ; encoding: [0x7b,0xb8,0x0a,0x7e]
+# GFX11-FAKE16: v_ceil_f16_e32 v5, ttmp15               ; encoding: [0x7b,0xb8,0x0a,0x7e]
 
-# GFX11-FAKE16: v_ceil_f16_e32 v5, m0                   ; encoding: [0x7d,0xb8,0x0a,0x7e]
-# GFX11-REAL16: v_ceil_f16_e32 v5.l, m0                 ; encoding: [0x7d,0xb8,0x0a,0x7e]
 0x7d,0xb8,0x0a,0x7e
+# GFX11-REAL16: v_ceil_f16_e32 v5.l, m0                 ; encoding: [0x7d,0xb8,0x0a,0x7e]
+# GFX11-FAKE16: v_ceil_f16_e32 v5, m0                   ; encoding: [0x7d,0xb8,0x0a,0x7e]
 
-# GFX11-FAKE16: v_ceil_f16_e32 v5, exec_lo              ; encoding: [0x7e,0xb8,0x0a,0x7e]
-# GFX11-REAL16: v_ceil_f16_e32 v5.l, exec_lo            ; encoding: [0x7e,0xb8,0x0a,0x7e]
 0x7e,0xb8,0x0a,0x7e
+# GFX11-REAL16: v_ceil_f16_e32 v5.l, exec_lo            ; encoding: [0x7e,0xb8,0x0a,0x7e]
+# GFX11-FAKE16: v_ceil_f16_e32 v5, exec_lo              ; encoding: [0x7e,0xb8,0x0a,0x7e]
 
-# GFX11-FAKE16: v_ceil_f16_e32 v5, exec_hi              ; encoding: [0x7f,0xb8,0x0a,0x7e]
-# GFX11-REAL16: v_ceil_f16_e32 v5.l, exec_hi            ; encoding: [0x7f,0xb8,0x0a,0x7e]
 0x7f,0xb8,0x0a,0x7e
+# GFX11-REAL16: v_ceil_f16_e32 v5.l, exec_hi            ; encoding: [0x7f,0xb8,0x0a,0x7e]
+# GFX11-FAKE16: v_ceil_f16_e32 v5, exec_hi              ; encoding: [0x7f,0xb8,0x0a,0x7e]
 
-# GFX11-FAKE16: v_ceil_f16_e32 v5, null                 ; encoding: [0x7c,0xb8,0x0a,0x7e]
-# GFX11-REAL16: v_ceil_f16_e32 v5.l, null               ; encoding: [0x7c,0xb8,0x0a,0x7e]
 0x7c,0xb8,0x0a,0x7e
+# GFX11-REAL16: v_ceil_f16_e32 v5.l, null               ; encoding: [0x7c,0xb8,0x0a,0x7e]
+# GFX11-FAKE16: v_ceil_f16_e32 v5, null                 ; encoding: [0x7c,0xb8,0x0a,0x7e]
 
-# GFX11-FAKE16: v_ceil_f16_e32 v5, -1                   ; encoding: [0xc1,0xb8,0x0a,0x7e]
-# GFX11-REAL16: v_ceil_f16_e32 v5.l, -1                 ; encoding: [0xc1,0xb8,0x0a,0x7e]
 0xc1,0xb8,0x0a,0x7e
+# GFX11-REAL16: v_ceil_f16_e32 v5.l, -1                 ; encoding: [0xc1,0xb8,0x0a,0x7e]
+# GFX11-FAKE16: v_ceil_f16_e32 v5, -1                   ; encoding: [0xc1,0xb8,0x0a,0x7e]
 
-# GFX11-FAKE16: v_ceil_f16_e32 v127, 0.5                ; encoding: [0xf0,0xb8,0xfe,0x7e]
-# GFX11-REAL16: v_ceil_f16_e32 v127.l, 0.5              ; encoding: [0xf0,0xb8,0xfe,0x7e]
 0xf0,0xb8,0xfe,0x7e
+# GFX11-REAL16: v_ceil_f16_e32 v127.l, 0.5              ; encoding: [0xf0,0xb8,0xfe,0x7e]
+# GFX11-FAKE16: v_ceil_f16_e32 v127, 0.5                ; encoding: [0xf0,0xb8,0xfe,0x7e]
 
-# COM: GFX11-FAKE16: warning: invalid instruction encoding 0xfd,0xb8,0x0a,0x7f
-# GFX11-REAL16: v_ceil_f16_e32 v5.h, src_scc            ; encoding: [0xfd,0xb8,0x0a,0x7f]
 0xfd,0xb8,0x0a,0x7f
+# GFX11-REAL16: v_ceil_f16_e32 v5.h, src_scc            ; encoding: [0xfd,0xb8,0x0a,0x7f]
```

Comment:
Why do we not have Fake16 output here? We have some output for most cases of using .h register input, but not all.
Othewise LGTM

Reply:
Yes that's the current behavior

also dicussed this here https://github.com/llvm/llvm-project/pull/115233#discussion_r1838538202

Reply:
I see, it's one of those cases. It is a shame bad input leads to to 3 very different types of output, but the conclusion will be the same, that it is not worth improving.